### DISCUSSION TITLE
Work around to handle multiple grouping in microsoft threat modeling template

### DISCUSSION THREAD

**Discussion by rsrinivasanhome** [simonec73/threatsmanager#19] [Q&A] [answered]
This is regarding the[ issue 17](https://github.com/simonec73/threatsmanager/issues/17) .  I want some advise on the workarounds possible here  . Previous in microsoft threat modeling tool we had the option to create a generic mobile app.  You can see the tool bar below.
![buttonsPNG](https://user-images.githubusercontent.com/31912418/147043998-99a16c4e-db4e-47e0-a137-52330c194ba6.PNG)

My rules were based on the generic mobile app which had its own set of properties .  The generic mobile app did not get imported into threats manager studio but my ios app  and android app have.  

Now i have thought as a workaround to change the schema definition of  ios app to incorporate the fields in generic mobile app  and similar with android app.  Then I need to update the rules for this type of change.
Is there any other option ?  Will derived template help here ?  Will it be able to work for item templates imported from microsoft threat modeling template ?

**Comment by simonec73** [comment, marked as answer]
My tool does consider the root nodes as unnecessary, because they typically correspond to the standard External Interactors, Processes, Data Stores, and so on. As such, they are not created. What you are showing me is that some templates have a different hierarchy. Ok, this means that I have to change the behavior of the TMT import function. I will do that, most probably for version 1.5.2.

Please note that this means that the **Representation** node in the TMT file will determine the type of object:
- **Rectangle** means External Interactor.
- **Ellipse** means Process.
- **ParallelLines** means Data Store.
- **BorderBoundary** means Trust Boundary.
- **LineBoundary** is ignored, because Line Trust Boundaries are not supported as already discussed in a previous message.
- **Line** means Flow.
- **Annotation** is ignored, because it is not possible to understand where the note should be applied.

All of those values can be applied to the Items in the Template Editor.

Would that work?

**Comment by rsrinivasanhome** [comment]
Thanks for the information it is very useful .  Need one input.

In today's  scenario is this the best work around for the problem I am facing.

In the schema definition of ios app to incorporate the fields of generic mobile app and perform similar changes with android app. Then I need to update the rules for this type of change.

Is there any other option ? Will derived template help here ? Will it be able to work for item templates imported from microsoft threat modeling template ?

**Reply by simonec73** [reply]
I am sorry. I do not get it. Can you rephrase, please?

**Reply by rsrinivasanhome** [reply]
I think I have an idea to solve the problem .  I will let you know if I need any inputs.

**Reply by rsrinivasanhome** [reply]
In the template that I have imported my Android App has become a Process .(Although I think it should be an external entity)
![image](https://user-images.githubusercontent.com/31912418/147200566-41119fcf-8f10-441a-a985-4be0ec063b4f.png)

Since it is a process it is getting many unnecessary process related attribute.   
Is there any way to remove the process related schema from all instance of the android app. (Not impacting my other process item templates)
I have tried remove property schema but it removes the attribute for that particular instance alone


I have some common properties for my ios and android app.  I tried creating a new property schema called generic mobile application with the common properties.

But  when I use apply property schema I am able to add the properties to that particular instance of ios app but not all instances.
Is entering the same properties for ios and android app the only option ?

One more question 
While defining the property schema can I define a default value for a set of values ?
![image](https://user-images.githubusercontent.com/31912418/147203886-60ac8378-240a-4800-ada0-159e8983c291.png)

**Reply by simonec73** [reply]
I am not sure how to answer. All of your struggles seem to depend from a bad source. This should be fixed. It does not make much sense to try and fix the tool because of a bad approach in creating templates.

That said, as I have previously written, there is something I can do and I will try to help as much as I can, but again if an entity is marked as Process I cannot do anything else than treating it as a Process. That's just an example, of course.

Consider though that you can add or remove a Property Schema from any object by using the context menu: there are specific actions you can use, introduced with v1.5.0.

**Reply by simonec73** [reply]
The answer to your question: can you set a default value? The answer is no: the default value is blank for text and false for booleans.

**Reply by rsrinivasanhome** [reply]
This particular case I can change it in the microsoft threat modeling tool in notepad and re - import.  The bad source could be fixed.

We have only a fixed set of item template types , data store , process , external entity , Flow, boundary there could be scenarios where all properties of an item type don`t make sense for all item templates.  In the microsoft template we can define defaults for properties specific to a particular item template and mark them as "Not Applicable" or assign a default.

**Comment by rsrinivasanhome** [comment]
To fix this out .  I tried to make the changes in a microsoft threat model using notepad.  It opens fine and appears correctly after importing in threats manage studio .  Then I created a template out of it  just containing the item templates that need correction.  

I took my original threat manger studio threat model (which had the issue) and then clicked import and applied the new template.  
All item templates which were previously used in the threat model do not get updated properly while others do.  (I even tried removing  all the item templates from the model but it did not help.)

The merge feature was however able to make the correction.  Now my android app is an external interactor.  Thanks for you inputs on the merge feature.

**Reply by simonec73** [reply]
No problem.
